test/gtest/bn_bwd.cpp

```diff
@@ -26,46 +26,123 @@
 
 #include "bn.hpp"
 
-struct GPU_BNBwd_FP16
+// https://github.com/ROCm/MIOpen/issues/1549
+// NCHW solver accepts
+// XDataType       : half_float::half
+// YDataYype       : half_float::half
+// ScaleDataType   : half_float::half
+// BiasDataType    : half_float::half
+// MeanVarDataType : half_float::half
+// struct GPU_BN_V1_BwdNCHW_FP16 : BNBwdTest<half_float::half, half_float::half, half_float::half,
+// half_float::half, half_float::half, half_float::half, half_float::half>
+// {
+// };
+
+// NHWC solver accepts
+// XDataType       : half_float::half
+// YDataYype       : half_float::half
+// ScaleDataType   : half_float::half
+// BiasDataType    : half_float::half
+// MeanVarDataType : float
+struct GPU_BN_V2_BwdNHWC_FP16
     : BNBwdTest<half_float::half, float, float, float, half_float::half, float, float>
 {
 };
 
-struct GPU_BNBwd_FP32 : BNBwdTest<float, float, float, float, float, float, float>
+// bf16 NHWC solver accepts is only on CK solver
+// XDataType       : bfloat16
+// YDataYype       : bfloat16
+// ScaleDataType   : bfloat16
+// BiasDataType    : bfloat16
+// MeanVarDataType : float
+struct GPU_BN_V1_BwdNHWC_BFP16 : BNBwdTest<bfloat16, float, float, float, bfloat16, float, float>
+{
+};
+
+struct GPU_BN_V2_BwdNHWC_BFP16 : BNBwdTest<bfloat16, float, float, float, bfloat16, float, float>
+{
+};
+
+struct GPU_BN_V1_Bwd_FP32 : BNBwdTest<float, float, float, float, float, float, float>
+{
+};
+
+struct GPU_BN_V2_Bwd_FP32 : BNBwdTest<float, float, float, float, float, float, float>
 {
 };
 
-struct GPU_BNBwd_BFP16 : BNBwdTest<bfloat16, float, float, float, bfloat16, float, float>
+struct GPU_BN_V1_BwdNHWC_FP64 : BNBwdTest<double, double, double, double, double, double, double>
 {
 };
 
-struct GPU_BNBwd_FP64 : BNBwdTest<double, double, double, double, double, double, double>
+struct GPU_BN_V2_BwdNHWC_FP64 : BNBwdTest<double, double, double, double, double, double, double>
 {
 };
 
-TEST_P(GPU_BNBwd_FP16, BnBwdCKHalf) {}
+// fp16
+// TEST_P(GPU_BN_V1_BwdNCHW_FP16, BnV1BwdHalf) {}
+TEST_P(GPU_BN_V2_BwdNHWC_FP16, BnV2BwdCKHalf) {}
+
+// float
+TEST_P(GPU_BN_V1_Bwd_FP32, BnV1BwdFloat) {}
+TEST_P(GPU_BN_V2_Bwd_FP32, BnV2BwdFloat) {}
+
+// bfp16 is only on CK solver
+TEST_P(GPU_BN_V1_BwdNHWC_BFP16, BnV1BwdCKBfloat) {}
+TEST_P(GPU_BN_V2_BwdNHWC_BFP16, BnV2BwdCKBfloat) {}
 
-TEST_P(GPU_BNBwd_FP32, BnBwdCKFloat) {}
+// double is only on CK solver
+TEST_P(GPU_BN_V1_BwdNHWC_FP64, BnV1BwdCKDouble) {}
+TEST_P(GPU_BN_V2_BwdNHWC_FP64, BnV2BwdCKDouble) {}
 
-TEST_P(GPU_BNBwd_BFP16, BnBwdCKBFloat16) {}
-TEST_P(GPU_BNBwd_FP64, BnBwdCKDouble) {}
+// // fp16
+// INSTANTIATE_TEST_SUITE_P(Smoke,
+//                          GPU_BN_V1_BwdNCHW_FP16,
+//                          testing::Combine(testing::ValuesIn(NetworkSmall<BNTestCase>()),
+//                                           testing::Values(miopenTensorNCHW),
+//                                           testing::ValuesIn({testBNAPIV1})));
```

Comment:
Any reason why this one is commented out?

Reply:
I commented this since the ocl batch norm solver NCHW is not working. I think we need to fix that but that is separate task. I have commented for now so that we don't forget this issue.